src/Tribe/Template.php

```diff
@@ -519,19 +533,32 @@ public function template( $name, $context = array(), $echo = true ) {
 		// Setup the Hook name
 		$hook_name = implode( '/', $namespace );
 
-		// Check if the file exists
-		$file = $this->get_template_file( $name );
+		// Cache file location and existence.
+		if ( ! isset( $file_exists[ $cache_name_key ] ) || ! isset( $files[ $cache_name_key ] ) ) {
+			// Check if the file exists
+			$files[ $cache_name_key ] = $file = $this->get_template_file( $name );
 
-		// Check if it's a valid variable
-		if ( ! $file ) {
-			return false;
+			// Check if it's a valid variable
+			if ( ! $file ) {
+				return $file_exists[ $cache_name_key ] = false;
+			}
+
+			// Before we load the file we check if it exists
```

Comment:
:no_entry_sign: Inline comments must end in full-stops, exclamation marks, or question marks

<sub>hash: f9365072d67b1f6e2ac401b7786135e7</sub>